### PR TITLE
refactor map macro for more general use

### PR DESCRIPTION
**Pull request by yjhmelody**
Signed-off-by: yjhmelody <465402634@qq.com>

### CODE REVIEW

File: core/primitives/src/lib.rs [hunk 1]
```diff
@@ -26,11 +26,17 @@
 /// Can be used to create a `HashMap`.
 #[macro_export]
 macro_rules! map {
-	($( $name:expr => $value:expr ),*) => (
+    ($( $name:expr => $value:expr ),* ,) => (
```
Comment (xlc) [inline core/primitives/src/lib.rs]: can use ` $(,)?` to handle optional trailing comma

Comment (yjhmelody) [reply core/primitives/src/lib.rs]: Oh, I forget the syntax.

**Comment by bkchr** [comment]
Can you please explain the reasoning behind your changes? I don't see how that generalizes the macro.

**Comment by thiolliere** [comment]
> Can you please explain the reasoning behind your changes? I don't see how that generalizes the macro.

That just allow to declare with a trailing comma

**Comment by bkchr** [comment]
> > Can you please explain the reasoning behind your changes? I don't see how that generalizes the macro.
> 
> That just allow to declare with a trailing comma

The trailing comma was added after there was a suggestion in this pr. :woman_shrugging:

**Comment by thiolliere** [comment]
> > > Can you please explain the reasoning behind your changes? I don't see how that generalizes the macro.
> > 
> > 
> > That just allow to declare with a trailing comma
> 
> The trailing comma was added after there was a suggestion in this pr. woman_shrugging

no before the fix trailing comma was made mandatory

**Comment by bkchr** [comment]
I checked now the first commit and understand it. I just saw in the diff the the optional trailing comma was added.

**Comment by bkchr** [comment]
@yjhmelody status?

**Pull request merged**